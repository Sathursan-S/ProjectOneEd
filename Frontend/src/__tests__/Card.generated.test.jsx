import renderer from 'react-test-renderer';
import React from 'react';
import { FaStar } from 'react-icons/fa';

import { Link } from 'react-router-dom';
import classImg from '../../Images/Card.png';
import Card from '../Components/Card/Card';

jest.mock('react-icons/fa');
jest.mock('./Card.css');
jest.mock('react-router-dom');
jest.mock('../../Images/Card.png');

const renderTree = tree => renderer.create(tree);
describe('<Card>', () => {
  it('should render component', () => {
    expect(renderTree(<Card 
    />).toJSON()).toMatchSnapshot();
  });
  
});