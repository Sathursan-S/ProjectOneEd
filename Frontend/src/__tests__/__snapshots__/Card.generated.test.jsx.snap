// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> should render component 1`] = `
<div
  className="card-container"
>
  <div
    className="card-header"
  >
    <h1>
      Explore classes on SpaceEd
    </h1>
  </div>
  <div
    className="class-card-container"
  >
    <div
      className="class-card-content"
    >
      <img
        alt="cardimage"
        src="test-file-stub"
      />
      <div
        className="card-details"
      >
        <div
          className="card-details-1"
        >
          <span
            className="card-subject"
          >
            Mathematics
          </span>
          <span
            className="card-grade"
          >
            Grade 10
          </span>
          <span
            className="card-teacher"
          >
            Mr. Smith
          </span>
        </div>
        <hr />
        <div
          className="card-details-2"
        >
          <div
            className="card-enroll"
          >
            <span>
              Enrolled Students
            </span>
            <span>
              50
            </span>
          </div>
          <div
            className="card-medium"
          >
            <span>
              Medium
            </span>
            <span>
              English
            </span>
          </div>
        </div>
        <hr />
        <div
          className="card-details-3"
        >
          <span>
            Monthly fee
          </span>
          <span>
            $50/month
          </span>
        </div>
        <hr />
        <div
          className="card-details-4"
        >
          <span>
            An error occurred. Either the engine you requested does not exist or there was another issue processing your request. If this issue persists please contact us through our help center at help.openai.com.
          </span>
        </div>
      </div>
      <div
        className="card-button"
      >
        <div
          className="card-stars"
        />
      </div>
    </div>
    <div
      className="class-card-content"
    >
      <img
        alt="cardimage"
        src="test-file-stub"
      />
      <div
        className="card-details"
      >
        <div
          className="card-details-1"
        >
          <span
            className="card-subject"
          >
            Science
          </span>
          <span
            className="card-grade"
          >
            Grade 8
          </span>
          <span
            className="card-teacher"
          >
            Ms. Johnson
          </span>
        </div>
        <hr />
        <div
          className="card-details-2"
        >
          <div
            className="card-enroll"
          >
            <span>
              Enrolled Students
            </span>
            <span>
              40
            </span>
          </div>
          <div
            className="card-medium"
          >
            <span>
              Medium
            </span>
            <span>
              English
            </span>
          </div>
        </div>
        <hr />
        <div
          className="card-details-3"
        >
          <span>
            Monthly fee
          </span>
          <span>
            $40/month
          </span>
        </div>
        <hr />
        <div
          className="card-details-4"
        >
          <span>
            An error occurred. Either the engine you requested does not exist or there was another issue processing your request. If this issue persists please contact us through our help center at help.openai.com.
          </span>
        </div>
      </div>
      <div
        className="card-button"
      >
        <div
          className="card-stars"
        />
      </div>
    </div>
    <div
      className="class-card-content"
    >
      <img
        alt="cardimage"
        src="test-file-stub"
      />
      <div
        className="card-details"
      >
        <div
          className="card-details-1"
        >
          <span
            className="card-subject"
          >
            Science
          </span>
          <span
            className="card-grade"
          >
            Grade 8
          </span>
          <span
            className="card-teacher"
          >
            Ms. Johnson
          </span>
        </div>
        <hr />
        <div
          className="card-details-2"
        >
          <div
            className="card-enroll"
          >
            <span>
              Enrolled Students
            </span>
            <span>
              40
            </span>
          </div>
          <div
            className="card-medium"
          >
            <span>
              Medium
            </span>
            <span>
              English
            </span>
          </div>
        </div>
        <hr />
        <div
          className="card-details-3"
        >
          <span>
            Monthly fee
          </span>
          <span>
            $40/month
          </span>
        </div>
        <hr />
        <div
          className="card-details-4"
        >
          <span>
            An error occurred. Either the engine you requested does not exist or there was another issue processing your request. If this issue persists please contact us through our help center at help.openai.com.
          </span>
        </div>
      </div>
      <div
        className="card-button"
      >
        <div
          className="card-stars"
        />
      </div>
    </div>
    <div
      className="class-card-content"
    >
      <img
        alt="cardimage"
        src="test-file-stub"
      />
      <div
        className="card-details"
      >
        <div
          className="card-details-1"
        >
          <span
            className="card-subject"
          >
            Science
          </span>
          <span
            className="card-grade"
          >
            Grade 8
          </span>
          <span
            className="card-teacher"
          >
            Ms. Johnson
          </span>
        </div>
        <hr />
        <div
          className="card-details-2"
        >
          <div
            className="card-enroll"
          >
            <span>
              Enrolled Students
            </span>
            <span>
              40
            </span>
          </div>
          <div
            className="card-medium"
          >
            <span>
              Medium
            </span>
            <span>
              English
            </span>
          </div>
        </div>
        <hr />
        <div
          className="card-details-3"
        >
          <span>
            Monthly fee
          </span>
          <span>
            $40/month
          </span>
        </div>
        <hr />
        <div
          className="card-details-4"
        >
          <span>
            An error occurred. Either the engine you requested does not exist or there was another issue processing your request. If this issue persists please contact us through our help center at help.openai.com.
          </span>
        </div>
      </div>
      <div
        className="card-button"
      >
        <div
          className="card-stars"
        />
      </div>
    </div>
    <div
      className="class-card-content"
    >
      <img
        alt="cardimage"
        src="test-file-stub"
      />
      <div
        className="card-details"
      >
        <div
          className="card-details-1"
        >
          <span
            className="card-subject"
          >
            Science
          </span>
          <span
            className="card-grade"
          >
            Grade 8
          </span>
          <span
            className="card-teacher"
          >
            Ms. Johnson
          </span>
        </div>
        <hr />
        <div
          className="card-details-2"
        >
          <div
            className="card-enroll"
          >
            <span>
              Enrolled Students
            </span>
            <span>
              40
            </span>
          </div>
          <div
            className="card-medium"
          >
            <span>
              Medium
            </span>
            <span>
              English
            </span>
          </div>
        </div>
        <hr />
        <div
          className="card-details-3"
        >
          <span>
            Monthly fee
          </span>
          <span>
            $40/month
          </span>
        </div>
        <hr />
        <div
          className="card-details-4"
        >
          <span>
            An error occurred. Either the engine you requested does not exist or there was another issue processing your request. If this issue persists please contact us through our help center at help.openai.com.
          </span>
        </div>
      </div>
      <div
        className="card-button"
      >
        <div
          className="card-stars"
        />
      </div>
    </div>
  </div>
</div>
`;
